### ISSUE TITLE
Can't display thai Subtitles

### ISSUE THREAD

**Issue by strikerhaven** [peno64/service.subtitles.opensubtitles-com-dualsub#2]
To make kodi display thai subs I needed to copy over a different font set otherwise it displayed rectangles. When I use the dual subtitles addon, I cannot change the font to the one that supports thai subtitiles and it just displays the rectangles. Can you please allow the dual subs addon to use same fonts as the default kodi subtitles display or allow me to pick a custom font from a file. Thank you.

**Comment by peno64** [comment]
You can change fonts via the settings of the addon

**Comment by strikerhaven** [comment]
> You can change fonts via the settings of the addon

Yes, first thing I tried, none of the fonts present resolved the issue. I need to be able to add custom fonts which I downloaded and manually added to kodi. I use coreelec and libreelec not windows BTW. The way i fixed it for single subs was to copy the sub files over to a specific location on the file system. Can you amend the addon so you can browse for custom fonts? Or use the same the the default system uses as I have already installed the correct font that supports thai. Failing that if there is another way to allow display of thai fonts please direct me to instructions as I can find none specific to the addon. Thank you.

**Comment by peno64** [comment]
Well in fact in the addon you dont choose a font but a character set. I guess the (any) addon uses the default kodi font and as far as I know there is no way to choose a font via an addon. 
However the addon is using the subtitle format .ass to be able to display two subs and maybe in that format it is possible to specify a font. I must investigate this.
Can you provide me a subtitle file and the font file to display thise characters? I also have libreelec on raspberry pi so that I can test this.

**Comment by strikerhaven** [comment]
Thanks for your assistance, btw I really liked the way your addon works
with the subs at the top and the bottom.

I think when you change the font type in kodi to the one attached then thai
subs work correctly. You need to place this is a specific location on the
libreelec filesystem and then reboot at this font appears as an option.

**Comment by peno64** [comment]
You didn't attach anything...

**Comment by strikerhaven** [comment]
Preview attachment AD-TT-Garuda-Bold.ttf
AD-TT-Garuda-Bold.ttf
75 KB
<https://mail.google.com/mail/u/0?ui=2&ik=3db1d43127&attid=0.1&permmsgid=msg-a:r1842667598570926595&th=18ea7c5bcb27c4a1&view=att&disp=safe&realattid=f_lukukuau0>

I can see in it in the message thread, strange that you cannot. does it
show this time?

**Comment by peno64** [comment]
No I can't see any attachments. I can see that you send your messages via mail and not via github itself. So probably they are in your gmail but I can't see them.

**Comment by peno64** [comment]
Looking a bit further into this.
In the addon settings you can choose a font and you can choose a character set.
For the font selection there are only 3 options atm. Arial, Vendana and DejaVu Sans.
Before I add a possibility to choose another font, have you tried to change the character set? First try "Auto Chardet", then "Auto_Charset_normalizer" and finally also try Thai.

**Comment by strikerhaven** [comment]
I got an error saying github wont accept as too large sorry the subtitle
just won't seem to go through.

**Comment by peno64** [comment]
@strikerhaven 
I added in the settings of the addon the possibility to choose as font <Kodi Subtitles Font>
When selected, the font selected in Kodi settings, Player, Subtitles, Font will be used. Note that you must have selected a specific font here for this to work. If that setting is the default font then the addon will give a message and cannot work because it can not determine the font in that case (Kodi then always return as name Default)

**Comment by strikerhaven** [comment]
Thank you, I will give it a try and let you know how things go.

**Comment by strikerhaven** [comment]
It worked, they display correctly now, thank you for your time and deep
consideration.
